### ISSUE TITLE
Feature: Improve the Recycle Bin properties window

### ISSUE THREAD

**Issue by verdaderoken** [files-community/Files#10343]
### What feature or improvement do you think would benefit Files?

I just noticed Files would open Recycle Bin's properties as if it was a normal folder. 
The settings of the Recycle Bin should also be included in the redesign.

Files Recycle Bin properties window:
![image](https://user-images.githubusercontent.com/88200096/198904816-9713cc94-2bbd-43a2-9bd6-c631a2b5e063.png)

Default Recycle Bin properties window:
![image](https://user-images.githubusercontent.com/88200096/198904848-d63c3b27-160b-4e93-ad05-0c0116fb99ba.png)

### Requirements

- Have a Files app version of the Recycle Bin properties including these extra settings
![image](https://user-images.githubusercontent.com/88200096/198904993-208faa9f-032e-4568-b0c5-a7bfb42533b3.png)

- Temporarily redirect to a default Recycle Bin properties instead of the current one (If the above would take time to develop)

### Files Version

2.3.54.0 

### Windows Version

10.0.22623.875

**Comment by Josh-65** [comment]
Would be nice if the normal properties on it worked also. (If removing it was an idea) Then a new tab for the size limits.

**Comment by hecksmosis** [comment]
I think we could remove the Attributes and Created rows and just add a new page for the size limits

**Comment by hecksmosis** [comment]
@yair100 Is this ready to be worked on?

**Comment by yaira2** [comment]
> I think we could remove the Attributes and Created, Modified, Accessed rows and just add a new page for the size limits

I agree that more details can be hidden for recycle bin and replaced with an option to change the maximum size. We don't need as

I think a card with an option to change the maximum size should do the trick, the other options are app preferences and don't belong in the properties window.

![image](https://user-images.githubusercontent.com/39923744/212375805-b54bc0af-bdda-4196-a5ef-d0a6db708726.png)

**Comment by Josh-65** [comment]
That would have to show for all drive connected as they all have there own recycle bin folders

**Comment by yaira2** [comment]
I wasn't aware of that, we'll need a better design then.

**Comment by hecksmosis** [comment]
Are Recycle bin folders in other drives currently supported?

**Comment by Josh-65** [comment]
What do you mean by supported? All drive that have recycle bin support show in the recycle bin section and deleted items will go into like they should unless permanently delete option is checked.

**Comment by hecksmosis** [comment]
I mean in Files

**Comment by hecksmosis** [comment]
I have tested this and if you try to delete a file in a non-root physical drive it just deletes to the default C: Recycle Bin so we probably shouldn't implement the separate page until we do support the other bins (and instead use a single card for the time being)

**Comment by Josh-65** [comment]
Work fine for me. Deleting content on another drive will place it into the recycle bin folder of that drive. Are you using the $RECYCLE.BIN folder (Turn on showing protected system files) or the one on the sidebar. If its the sidebar then that just acts like a library showing all the drives together.

**Comment by hecksmosis** [comment]
Oh, I didn't know that

**Comment by hecksmosis** [comment]
Then we should implement the tab

**Comment by yaira2** [comment]
I'm not sure we need a tab, I still think a card with a list view would work. Could add a number box to the template for changing the size.

**Comment by 0x5bfa** [comment]
For others:

Those values are stored in the registry
- Maximum allowed Recycle Bin size: `Software\Microsoft\Windows\CurrentVersion\Policies\Explorer\RecycleBinSize`
- Do not move deleted files to the Recycle Bin:  `Software\Microsoft\Windows\CurrentVersion\Policies\Explorer\NoRecycleFiles`

As you may find, we have to follow this setting if we implement this page. And I agree with Yair, it’s not worth having a new page, just a card in General tab.

If no one here is interested in implementing this feature, I can work on. There’re some work left so it’ll take a time to get started; while then if someone come here and struggling I can help them.

**Comment by 0x5bfa** [comment]
> that just acts like a library showing all the drives together.

Current properties management sucks (SelectedItemPropertiesViewModel and BaseProperties classes). We should have single interface (as in IStorableProperties) and then inherit it from IFileProperties, IFolderProperties, IDriveProperties, IDeviceProperties, IShellFolderProperties, INetworkLocationProperties and IFileTagProperties.
Reflection can determine which pages should be shown and how properties should be changed via Details page or General page, using dedicated methods(as in ModifyLastAccess)
